### PR TITLE
chore: add both df tree impelementations to memory usage test

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

**Review by kostasrim — DISMISSED** [review]

File: src/core/bptree_set_test.cc [hunk 1]
```diff
@@ -302,11 +302,19 @@ TEST_F(BPTreeSetTest, MemoryUsage) {
   AllocType alloc(&mi_alloc);
   absl::btree_set<pair<double, sds>, std::greater<pair<double, sds>>, AllocType> btree(alloc);
 
-  LOG(INFO) << "btree before: " << mi_alloc.used() << " bytes";
+  ASSERT_EQ(0, mi_alloc.used());
   for (size_t i = 0; i < sds_vec.size(); ++i) {
     btree.emplace(i, sds_vec[i]);
   }
-  LOG(INFO) << "btree after: " << mi_alloc.used() << " bytes";
+  LOG(INFO) << "abseil btree: " << mi_alloc.used() << " bytes";
```
Comment (chakaz) [inline src/core/bptree_set_test.cc:309]:
```suggestion
  ASSERT_GT(mi_alloc.used(), 0) << mi_alloc.used();
```

Comment (chakaz) [reply src/core/bptree_set_test.cc:309]: (to test that `btree_set` actually uses `alloc`)

File: src/core/bptree_set_test.cc [hunk 2]
```diff
@@ -302,11 +302,19 @@ TEST_F(BPTreeSetTest, MemoryUsage) {
...
+  btree.clear();
+
+  ASSERT_EQ(0, mi_alloc.used());
+  SDSTree df_tree(&mi_alloc);
+  for (size_t i = 0; i < sds_vec.size(); ++i) {
+    btree.emplace(i, sds_vec[i]);
+  }
+  LOG(INFO) << "df btree: " << mi_alloc.used() << " bytes";
```
Comment (chakaz) [inline src/core/bptree_set_test.cc:317]: ditto

File: src/core/bptree_set_test.cc (hunk 1 above)
Comment (romange) [reply src/core/bptree_set_test.cc]: done

**Review by chakaz — APPROVED** [review]

**Pull request merged**